src/lib.rs

```diff
@@ -53,15 +43,12 @@ pub fn candidate_cert_dirs() -> impl Iterator<Item = &'static Path> {
     .filter(|p| p.exists())
 }
 
-/// Deprecated as this isn't sound, use [`init_openssl_env_vars`] instead.
-#[doc(hidden)]
-#[deprecated(note = "this function is not safe, use `init_openssl_env_vars` instead")]
-pub fn init_ssl_cert_env_vars() {
-    unsafe {
-        init_openssl_env_vars();
-    }
-}
-
+#[cfg(feature = "allow-unsafe-set-env")]
+/// WARNING: This method is likely not safe to use, unless 1) you are
+/// performing these actions at the start of your main function, or 2) you are
+/// otherwise absolutely certain that you are running in a single-threaded
+/// environment.
```

Comment:
There is already a `/// # Safety` comment, which is the idiomatic way of communicating this. If there's more to say about the safety invariant I think it should go there.

The language "likely not safe to use" is a bit wishy-washy. I think "is unsound unless" is more precise.

Also, (1) is not a separate case from (2); the start of `main` is just one example of a probably single-threaded environment. It seems out of place here to provide detailed guidance about how to meet that obligation.

Reply:
> There is already a /// # Safety comment, which is the idiomatic way of communicating this. If there's more to say about the safety invariant I think it should go there.
> The language "likely not safe to use" is a bit wishy-washy. I think "is unsound unless" is more precise.

Fair, I'll tweak.

> (1) is not a separate case from (2); the start of main is just one example of a probably single-threaded environment.

I prefer clarity over conciseness, even if it's redundant.

> It seems out of place here to provide detailed guidance about how to meet that obligation.

Given that the previous information was not enough to prevent a consumer from re-implementing and ignoring the safety information, I have to disagree.